components/database.py

```diff
@@ -206,116 +239,153 @@ def get_all_past_mounted_modules():
-                  "AND M2.acadYearAndSem LIKE %s" + \
-                  "ORDER BY m2.moduleCode, m2.acadYearAndSem"
-    processed_ay = selected_ay + "%"
-
-    DB_CURSOR.execute(sql_command, (processed_ay,))
-    return DB_CURSOR.fetchall()
+    try:
+        sql_command = "SELECT m2.moduleCode, m1.name, m2.acadYearAndSem, m2.quota " +\
+                      "FROM module m1, moduleMountTentative m2 WHERE m2.moduleCode = m1.code " +\
+                      "AND M2.acadYearAndSem LIKE %s" + \
+                      "ORDER BY m2.moduleCode, m2.acadYearAndSem"
+        processed_ay = selected_ay + "%"
+
+        DB_CURSOR.execute(sql_command, (processed_ay,))
+        return DB_CURSOR.fetchall()
+    except psycopg2.Error:
+        CONNECTION.rollback()
+        return []
 
 
 def get_fixed_mounting_and_quota(code):
     '''
         Get the fixed AY/Sem and quota of a mounted module
     '''
-    sql_command = "SELECT acadYearAndSem, quota FROM moduleMounted " +\
-                  "WHERE moduleCode=%s ORDER BY acadYearAndSem ASC"
-    DB_CURSOR.execute(sql_command, (code, ))
-    return DB_CURSOR.fetchall()
+    try:
+        sql_command = "SELECT acadYearAndSem, quota FROM moduleMounted " +\
+                      "WHERE moduleCode=%s ORDER BY acadYearAndSem ASC"
+        DB_CURSOR.execute(sql_command, (code, ))
+        return DB_CURSOR.fetchall()
+    except psycopg2.Error:
+        CONNECTION.rollback()
+        return []
 
 
 def get_tenta_mounting_and_quota(code):
     '''
         Get the tentative AY/Sem and quota of a mounted module
     '''
-    sql_command = "SELECT acadYearAndSem, quota FROM moduleMountTentative " +\
-                  "WHERE moduleCode=%s ORDER BY acadYearAndSem ASC"
-    DB_CURSOR.execute(sql_command, (code, ))
-    return DB_CURSOR.fetchall()
+    try:
+        sql_command = "SELECT acadYearAndSem, quota FROM moduleMountTentative " +\
+                      "WHERE moduleCode=%s ORDER BY acadYearAndSem ASC"
+        DB_CURSOR.execute(sql_command, (code, ))
+        return DB_CURSOR.fetchall()
+    except psycopg2.Error:
+        CONNECTION.rollback()
+        return []
 
 
 def get_mounting_of_target_fixed_ay_sem(code, ay_sem):
     '''
         Get the mounting status of a module in a target fixed AY/Sem
     '''
-    sql_command = "SELECT COUNT(*) FROM moduleMounted " +\
-                  "WHERE moduleCode=%s AND acadYearAndSem=%s"
-    DB_CURSOR.execute(sql_command, (code, ay_sem))
-    result = DB_CURSOR.fetchone()
-    return result[0] == 1    # True == Mounted, False == Not Mounted
-
+    try:
+        sql_command = "SELECT COUNT(*) FROM moduleMounted " +\
+                      "WHERE moduleCode=%s AND acadYearAndSem=%s"
+        DB_CURSOR.execute(sql_command, (code, ay_sem))
+        result = DB_CURSOR.fetchone()
+        return result[0] == 1    # True == Mounted, False == Not Mounted
+    except psycopg2.Error:
+        CONNECTION.rollback()
+        return []
```

Comment:
Note that the try block returns a boolean value, so the except clause should return a boolean value as well instead of an empty list.